frontend/taipy-gui/src/components/Taipy/Chat.tsx

```diff
@@ -256,30 +271,54 @@ const Chat = (props: ChatProps) => {
                 const elt = evt.currentTarget.querySelector("input");
                 if (elt?.value) {
                     dispatch(
-                        createSendActionNameAction(id, module, onAction, evt.key, updateVarName, elt?.value, senderId)
+                        createSendActionNameAction(id, module, onAction, evt.key, updateVarName, elt?.value,senderId, imagePreview)
                     );
                     elt.value = "";
+                    setSelectedFile(null);
+                    setImagePreview(null);
                 }
                 evt.preventDefault();
             }
         },
-        [updateVarName, onAction, senderId, id, dispatch, module]
+        [imagePreview, updateVarName, onAction, senderId, id, dispatch, module]
     );
 
     const handleClick = useCallback(
         (evt: MouseEvent<HTMLButtonElement>) => {
             const elt = evt.currentTarget.parentElement?.parentElement?.querySelector("input");
             if (elt?.value) {
                 dispatch(
-                    createSendActionNameAction(id, module, onAction, "click", updateVarName, elt?.value, senderId)
+                    createSendActionNameAction(id, module, onAction, "click", updateVarName, elt?.value,senderId,imagePreview)
                 );
                 elt.value = "";
+                setSelectedFile(null);
+                setImagePreview(null);
             }
             evt.preventDefault();
         },
-        [updateVarName, onAction, senderId, id, dispatch, module]
+        [imagePreview,updateVarName, onAction, senderId, id, dispatch, module]
     );
 
+    const handleFileSelect = (event: React.ChangeEvent<HTMLInputElement>) => {
+            const file = event.target.files ? event.target.files[0] : null;
+            if (file) {
+                if (file.type.startsWith("image/")) {
+                    setSelectedFile(file);
+                    setImagePreview(URL.createObjectURL(file));
```

Comment:
Should we limit file size ?

Reply:
and invoke [URL.revokeObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) when the URL is not needed anymore ?

Reply:
The chat component needs to display the images with the URLs after message submissions. When should I revoke them? 

I added `max_file_size` property to the Chat component. I will add the test for this and consider adding a warning about file type and size.

Reply:
When the component is unmounted?
In the returned function from use effect...

Reply:
I tried to implement but not sure if it's right. Please have a look at it.